warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java

```diff
@@ -603,6 +603,10 @@ public static Matcher compileFieldNamePattern(String fieldNamePattern) {
         
         if (types.isEmpty()) {
             types.addAll(typeFieldMap.get(null));
+        } else {
+            synchronized (typeFieldMap) {
```

Comment:
Not sure why locking on typeFieldMap is needed here. If it is needed, then read/writes to typeFieldMap elsewhere in this class would also need to be guarded, right?

Reply:
Also, locking on a field to guard read/write on the same field is generally not a good idea, since the lock itself corresponds to the referenced object, not to the field itself. E.g., if the field ever gets reassigned to a new/different instance, then the existing lock becomes worthless. Wouldn't be an issue here, since there's only ever one assignment and the field is private, but still probably better to lock on `this` or something that's final wrt to the current instance

Reply:
I'm not sure Keith's question re why is the locking needed in the first place has been addressed. If there is a need to lock typeFieldMap, then there are many race conditions in this class, including in the getDataTypes method that has been modified. The map is queried earlier in the method, but it is possible that between that check and where you are updating, the map could have been updated by a different thread. The local types array is then added to the map, but could it be different than one another thread put there?

Reply:
It appears it's now synchronized(this) instead of synchronized(typeFieldMap) .

Reply:
Yes, but the question of why the synchronization is needed in the first place is still outstanding. If that synchronization is truly necessary, then there are probably more serious problems in this class that need to be fixed.

Reply:
I'm not aware of any multi-threaded contexts for IngestHelpers, but I can't rule out that possibility. These are used primarily by ColumnBaseHandlers, which are leveraged during the map task, which *should* be single-threaded, unless we're doing something weird

Reply:
I think we can probably do without it. @ivakegg and I were discussing solutions and he threw this in quick without looking much at what was going on, but the more I look at it the less I think we need to synchronize here. @ivakegg concur?